src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs

```diff
@@ -17,6 +17,9 @@ public class SocketGuildChannel : SocketChannel, IGuildChannel
         public SocketGuild Guild { get; }
         public string Name { get; private set; }
         public int Position { get; private set; }
+        public ulong? ParentId { get; private set; }
+        public IGuildChannel ParentChannel => ParentId == null ? null : Guild.GetChannel(ParentId.Value);
+        public Task<IGuildChannel> GetParentChannelAsync() => Task.FromResult(ParentChannel);
```

Comment:
This should be an explicit interface implementation of `IGuildChannel` as it is redundant with the public property `ParentChannel`.

Reply:
Alongside, the type should be `IChannelCategory` as actual channels can't be parents/children of other channels, only categories.